status-im/status-desktop
# Refactor various popups

### What does the PR do

Refactors the various popups to remove logic, make them functional and simplifies the code. Also added storybook pages for each of these components.

### Affected areas

Popups affected:
- Send Contact Request Popup
- Review Contact Request Popup
- Remove Contact Popup
- Mark as Untrusted Popup
- Add Nickname Popup
- Block Contact popup
- Unblock Contact popup

Also affected but not testable right now because this feature is disabled (and likely will be removed):
- Contact Verification Popup
- Mark as ID Verified Popup
- Remove ID verification Popup
- Outgoing Contact Verification Request Popup

### Impact on end user

The behaviour of these popups should remain the same

### How to test

Generally, right clicking on the user profile and select the menu item, then test if the functionality still works.

example:
1. right click on a profile that is already a contact
2. click remove contact

expectation: the popup should contain the correct data about the user

3. click remove contact

expectation: contact should be removed

ID verification is less of a concern as this feature is disabled, no testing of this is expected.

### Risk 

Main risk is one of the popups listed in "Popups affected" no longer working.
